test/functions_test.py

```diff
@@ -78,24 +78,35 @@
 >>> random.seed(11)
 >>> sign_gen()
 -1
+>>> used_vertices = {k:[] for k in range(1,5)}
+>>> used_vertices = {k:[] for k in range(1,6)}
+>>> degree_dict = {1:2,2:3,3:3,4:3,5:3}
+>>> all_vertices = list(range(1, 6))
+>>> branch_gen(1,3,3,300,3000,1,2,2,1,all_vertices,used_vertices,degree_dict)
+[[], []]
```

Comment:
This test is not clear for me :confused: 
Shouldn't we fill the `used_vertices` with proper values so that `degree_dict` fits to it?